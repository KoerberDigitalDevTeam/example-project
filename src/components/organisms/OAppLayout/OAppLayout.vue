<template>
  <v-app v-testId.app-layout class="o-app-layout__app">
    <OMainMenu v-testId.app-layout.main-menu user-name="Admin" :menu-items="menuItems" />
    <v-main>
      <v-container
        v-testId.app-layout.page-content
        fluid
        class="o-app-layout__page-container"
        user-name="Admin"
      >
        <router-view></router-view>
      </v-container>
    </v-main>
  </v-app>
</template>

<script>
import OMainMenu from '@/components/organisms/OMainMenu/OMainMenu.vue';

export default {
  name: 'OAppLayout',

  components: {
    OMainMenu,
  },

  data() {
    return {
      menuItems: [
        {
          title: 'examplePage.title',
          path: '/',
          icon: 'mdi-domain',
        },
      ],
    };
  },
};
</script>

<style lang="scss" scoped>
::v-deep {
  .o-app-layout__page-container {
    height: 100%;
    max-height: 100vh;
    padding: 0;
  }
}
</style>
