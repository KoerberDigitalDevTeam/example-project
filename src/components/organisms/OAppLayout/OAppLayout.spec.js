import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
import OAppLayout from '@/components/organisms/OAppLayout/OAppLayout.vue';

Vue.use(VueRouter);

let vuetify;

function createWrapper(options = {}) {
  const wrapper = shallowMount(OAppLayout, {
    vuetify,
    ...options,
  });
  return wrapper;
}

beforeEach(() => {
  vuetify = new Vuetify();
});

describe('OAppLayout', () => {
  describe('Template:', () => {
    it('Should render a OAppLayout', () => {
      const wrapper = createWrapper();
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should have a section for PageContainer', () => {
      const wrapper = createWrapper();
      const pageContainerElement = wrapper.find('[data-testid=app-layout-page-content]');
      expect(pageContainerElement.exists()).toBe(true);
    });
  });
});
