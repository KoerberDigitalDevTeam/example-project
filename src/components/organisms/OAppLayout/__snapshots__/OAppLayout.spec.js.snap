// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OAppLayout Template: Should render a OAppLayout 1`] = `
<v-app-stub id="app" class="o-app-layout__app" data-testid="app-layout">
  <omainmenu-stub username="Admin" menuitems="[object Object]" data-testid="app-layout-main-menu"></omainmenu-stub>
  <v-main-stub tag="main">
    <v-container-stub tag="div" fluid="true" user-name="Admin" class="o-app-layout__page-container" data-testid="app-layout-page-content">
      <router-view-stub name="default"></router-view-stub>
    </v-container-stub>
  </v-main-stub>
</v-app-stub>
`;
